src/realm/util/file.cpp

```diff
@@ -198,6 +208,18 @@ void remove_dir_recursive(const std::string& path)
 }
 
 
+bool try_remove_dir_recursive(const std::string& path)
+{
+    try {
+        remove_dir_recursive(path);
+    }
+    catch (const File::NotFound&) {
+        return false;
+    }
+    return true;
+}
```

Comment:
In the interest of avoiding exceptions for normal control flow, I suggest the following alternative implementation:

    bool try_remove_dir_recursive(const std::string& path)
    {
        if (File::exists(path)) {
            remove_dir_recursive(path); // Throws
            return true;
        }
        return false;
    }

This has a built-in race condition (if the directory is removed between the check and the removal), but since `remove_dir_recursive()` has the same kind of race conditions, I think it is fine, at least for now.

One of the race conditions in `remove_dir_recursive()` is if a file is placed in the directory by another process or thread after we removed all the files, but before we remove the directory, then the removal will fail. Fixing this will require a retry mechanism, but it would have to be bounded to avoid indefinite "ping pong" with a file creating process / thread.

For now, I think it is better to just add a remark in the documentation of `File::remove_dir_recursive()` and `File::try_remove_dir_recursive()` that the functions may throw `AccessError` if another process or thread is interfering.